app/controllers/two_step_verification_exemptions_controller.rb

```diff
@@ -19,6 +22,13 @@ def update
 
 private
 
+  def parse_date_from_user_params(user_params)
+    year = user_params["expiry_date_for_2sv_exemption(1i)"].to_i
```

Comment:
What format is the date being passed through in? I'd expect to be able to do something like DateTime.parse(date) assuming it's in a consistent string

Reply:
So because the date comes from three form fields (as is given by the date_select in rails), it comes through in the shape of a year string, month string and day string. I _think_ DateTime.parse wouldn't be useful in this case since we'd have to artificially make it into a date time string,  but let me know if you can see a better way (also might be worth noting we're not using date times, but just dates)

Reply:
Had another think and have now changed this more along your lines to:
```    
date_part_params = ["expiry_date_for_2sv_exemption(1i)", "expiry_date_for_2sv_exemption(2i)", "expiry_date_for_2sv_exemption(3i)"]
    date_string = date_part_params.map { |param_name| user_params[param_name] }.join("/")
    Date.parse(date_string)
```

This also has the benefit of meaning that if one of the parts is ever not an integer (which should never happen in practice), it'll be caught by our existing rescue from an invalid date. `.to_i` is a bit sketchy.